RecoParticleFlow/PFClusterProducer/plugins/SpikeAndDoubleSpikeCleaner.cc

```diff
@@ -117,19 +117,53 @@ clean(const edm::Handle<reco::PFRecHitCollection>& input,
     }    
     const spike_cleaning& clean = _thresholds.find(hitlayer)->second;    
     if( rechit.energy() < clean._singleSpikeThresh ) continue;
+
+    //Fix needed for HF.  Here, we find the (up to) five companion rechits 
+    //to work in conjunction with the neighbours4() implementation below for the full HF surrounding energy
+    float compsumE = 0.0;
+    if ((hitlayer==PFLayer::HF_EM || hitlayer==PFLayer::HF_HAD)) {
+      int comp = 1;
+      if (hitlayer==PFLayer::HF_EM) comp = 2;
+      const HcalDetId& detid = (HcalDetId)rechit.detId();
+      int heta = detid.ieta();
+      int hphi = detid.iphi();
+      
+      //At eta>39, phi segmentation changes 
+      int predphi =2;
+      if (std::abs(heta)>39) predphi=4;
+
+      int curphiL = hphi-predphi;
+      int curphiH = hphi+predphi;
+
+      //HcalDetId valid phi range (1-72)
+      while (curphiL<0) curphiL+=72;
+      while (curphiH>72) curphiH-=72;
+
+      std::pair<std::vector<int>, std::vector<int>>  phietas({heta,heta+1,heta-1,heta,heta},{hphi,hphi,hphi,curphiL,curphiH});
+
+      std::vector<uint32_t> rawDetIds;
+      for(int in=0;in<5;in++) {
```

Comment:
please receive the length of the vector (5) - but better yet, don't hardwire 5, ask the vectors in the pair how many elements they have.